source/session.ts

```diff
@@ -77,18 +77,65 @@ export interface SearchOptions {
   objectTypeIds?: string[];
 }
 
-export interface Response<T> {
+interface ActionResponseMap {
+  query: {
+    data: Data[];
+  };
+  create: {
+    data: Data;
+  };
+  update: {
+    data: Data;
+  };
+  delete: {
+    data: boolean;
+  };
+  search: {
+    data: Data[];
+  };
+}
+
+interface BaseResponse<T = Data> {
   url?: any;
   headers?: any;
-  action: string;
+  action: keyof ActionResponseMap;
   metadata: {
     next: {
       offset: number | null;
     };
   };
+  data: T | T[] | boolean;
+}
+
+export interface QueryResponse<T = Data> extends BaseResponse<T> {
   data: T[];
+  action: "query";
 }
 
+export interface CreateResponse<T = Data> extends BaseResponse<T> {
+  data: T;
```

Comment:
set `url` and `headers` props to `never` for create, query, update, delete, and search